### PR TITLE
Implement delete button on versionchooser

### PR DESCRIPTION
**Pull request by Williangalvani**
an image for testing is available at williangalvani/companion-core:delete

### CODE REVIEW

File: core/services/versionchooser/static/index.html [hunk 1]
```diff
@@ -286,6 +287,22 @@ <h3>Applying...</h3>
...
+          const [repository, tag] = fullname.split(':')
+          axios({
+            method: 'delete',
+            url: '/v1.0/version/delete',
+            data: {
+              repository,
+              tag,
+            },
+          }).then(() => {
+            this.availableVersions['local'] = this.availableVersions['local'].filter(
```
Comment (patrickelectric) [inline core/services/versionchooser/static/index.html:300]: isn't it better to call `loadAvailableversions()` ? Or is there a specific reason to update `availableVersions` ? If so, could you comment it ?

File: core/services/versionchooser/utils/chooser.py [hunk 2]
```diff
@@ -28,32 +28,41 @@ def index() -> web.FileResponse:
         """Serve index.html"""
         return web.FileResponse(str(STATIC_FOLDER) + "/index.html", headers={"cache-control": "no-cache"})
 
+    @staticmethod
+    def get_current_image_and_tag() -> Optional[Tuple[str, str]]:
```
Comment (patrickelectric) [inline core/services/versionchooser/utils/chooser.py:32]: missing docstrings about each tuple field.

File: core/services/versionchooser/static/index.html (hunk 1 above)
Comment (Williangalvani) [reply core/services/versionchooser/static/index.html:300]: loadAvailableVersions takes a while, fetching from dockerhub can take some time. this is instantaneous, I'll add a comment.

**Comment by patrickelectric** [comment]
We should disable the delete button while waiting for a response. If someone click in delete twice a weird message appers:

image

Also, this appears to be a second bug since the message is not clear.

File: core/services/versionchooser/utils/chooser.py [hunk 3]
```diff
@@ -130,6 +130,49 @@ async def set_version(self, image: str, tag: str) -> web.StreamResponse:
...
+        """
+        full_name = f"{image}:{tag}"
+        # refuse if it is the current image
+        with open(DOCKER_CONFIG_PATH, "r+") as startup_file:
+            try:
+                core = json.load(startup_file)["core"]
+                current_tag = core["tag"]
+                current_image = core["image"]
+                if image == current_image and tag == current_tag:
+                    return web.Response(status=403, text=f"Refusing to delete the image in use ({image}:{tag})")
```
Comment (rafaellehmkuhl) [inline core/services/versionchooser/utils/chooser.py]: Maybe something like:

"Image 'image:tag' is in use and cannot be deleted."

File: core/services/versionchooser/utils/chooser.py [hunk 4]
```diff
@@ -130,6 +130,49 @@ async def set_version(self, image: str, tag: str) -> web.StreamResponse:
...
+            except Exception as e:
+                logging.warning(f"Unable to read config file: {e}")
+                return web.Response(status=500, text=f"Unable read config file: {e}")
+
```
Comment (rafaellehmkuhl) [inline core/services/versionchooser/utils/chooser.py]: "Unable *to read..."

File: core/services/versionchooser/utils/chooser.py [hunk 5]
```diff
@@ -130,6 +130,49 @@ async def set_version(self, image: str, tag: str) -> web.StreamResponse:
...
+        # check if image exists
+        image = self.client.images.get(full_name)
+        if not image:
+            logging.warning(f"Image not found: {full_name}")
+            return web.Response(status=404, text=f"image not found: {full_name}")
```
Comment (rafaellehmkuhl) [inline core/services/versionchooser/utils/chooser.py:165]: "Image 'full_name' not found".

File: core/services/versionchooser/utils/chooser.py [hunk 6]
```diff
@@ -130,6 +130,49 @@ async def set_version(self, image: str, tag: str) -> web.StreamResponse:
...
+
+        # actually attempt to delete it
+        logging.info(f"Deleting image {image}:{tag}...")
+        try:
+            self.client.images.remove(full_name, force=False, noprune=False)
+            logging.info("Image deleted successfully")
```
Comment (rafaellehmkuhl) [inline core/services/versionchooser/utils/chooser.py:171]: I think "Image successfully deleted." is the correct form.

**Review by rafaellehmkuhl — APPROVED** [review]

File: core/services/versionchooser/utils/chooser.py (hunk 6 above)
Comment (Williangalvani) [reply core/services/versionchooser/utils/chooser.py:171]: "succesfully" is an adverb of mode. Same as in portuguese, [they can be placed before and after the verb](https://www.twinkl.com.br/teaching-wiki/adverb-of-manner#:~:text=An%20adverb%20of%20manner%20modifies,the%20reader%20a%20clearer%20picture.).
This way sounds better to me.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**